<launch>
    <!-- Launch the simulation environment first -->
    <include file="$(find minilab_simulation)/launch/minilab_simulation.launch"/>

    <!-- SLAM avec gmapping -->
    <include file="$(find minilab_simulation)/launch/gmapping.launch"/>

    <!-- Launch teleop node for converting joy commands to robot velocity -->
    <node pkg="minilab_teleop" type="teleop_node.py" name="minilab_teleop" output="screen">
        <param name="linear_scale" value="0.5" />
        <param name="angular_scale" value="1.0" />
    </node>

    <!-- Launch joy_node -->
    <node name="joy_node" pkg="joy" type="joy_node"/>

    <!-- Planificateur de chemin -->
    <node name="path_planner" pkg="minilab_navigation" type="path_planner_node.py" output="screen"/>

    <!-- Waypoint follower avec les nouveaux paramètres -->
    <node name="waypoint_follower" pkg="minilab_navigation" type="waypoint_follower_node.py" output="screen">
        <!-- Paramètres géométriques du robot -->
        <param name="l1" value="0.15" />         <!-- Distance du point de contrôle à l'axe des roues -->
        
        <!-- Gains de la loi de commande -->
        <param name="k1" value="1.0" />         <!-- Gain pour l'erreur en x -->
        <param name="k2" value="1.0" />         <!-- Gain pour l'erreur en y -->
        
        <!-- Paramètres de sécurité -->
        <param name="max_linear_speed" value="0.5" />
        <param name="max_angular_speed" value="0.5" />
        <param name="dist_threshold" value="0.25" />
        
        <!-- Debug -->
        <param name="debug" value="true" />
    </node>

    <!-- RViz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find minilab_navigation)/config/path.rviz"/>
</launch>